File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Actions/Operation/FacilitySelectAndSet.cs
using Cen.Wms.Client.Actions.Data.Settings;
using Cen.Wms.Client.Actions.UI.Settings;
using Cen.Wms.Client.Models.State;
using Cen.Wms.Client.Services;

namespace Cen.Wms.Client.Actions.Operation
{
    class FacilitySelectAndSet
    {
        public static bool Run()
        {
            var result = false;

            var selectedFacilityId = FacilitySelect.Run();
            if (selectedFacilityId != null)
            {
                var facilityConfig = FacilityConfigGet.Run(selectedFacilityId);
                if (facilityConfig != null)
                {
                    var settingsFacility = new SettingsFacility
                    {
                        FacilityId = selectedFacilityId,
                        AcceptanceProcessType = facilityConfig.AcceptanceProcessType,
                        PalletCodePrefix = facilityConfig.PalletCodePrefix,
                        IsAcceptanceByPapersEnabled = facilityConfig.IsAcceptanceByPapersEnabled
                    };

                    GStateProvider.Instance.SetSettingsFacility(settingsFacility);
                    result = true;
                }
            }

            return result;
        }
    }
}

File: TCDNew-master/src/ui.android/lib/actions/operation/FacilitySelectAndSet.dart
import 'package:flutter/widgets.dart';
import 'package:tiq/actions/data/settings/FacilityConfigGet.dart';
import 'package:tiq/actions/ui/settings/FacilitySelect.dart';
import 'package:tiq/models/state/SettingsFacility.dart';
import 'package:tiq/services/GStateProvider.dart';

class FacilitySelectAndSet {
  Future<bool> run(BuildContext context) async {
    var result = false;

    final selectedFacilityId = await FacilitySelect().run(context);
    if (selectedFacilityId != null) {
      final facilityConfig = await FacilityConfigGet().run(context, selectedFacilityId);
      if (facilityConfig != null) {
        final settingsFacility = SettingsFacility(
          selectedFacilityId,
          facilityConfig.acceptanceProcessType,
          facilityConfig.palletCodePrefix,
          facilityConfig.isAcceptanceByPapersEnabled
        );

        GStateProvider.instance.setSettingsFacility(settingsFacility);
        result = true;
      }
    }

    return result;
  }
}
